### DISCUSSION TITLE
Orgmode-like Heading Blocks

### DISCUSSION THREAD

**Discussion by Thyverity** [suchnsuch/Tangent#70] [Ideas]
In both Obsidian and Tangent, the page is the smallest unit of information. While this is fine, pages are quite non-granular, which limits the possibilities for interaction. On the other hand, in Logseq, the smallest unit is the block (line), which provides much greater flexibility.
Can this be replicated within Tangent in a similar way? perhaps a balanced approach with headings as the smallest unit, orgmode style.

Where's a list of features that heading blocks would, if at least, influence:
- Heading properties and tags, subheadings would inherit the parent's properties and tags.
- Folding headings
- Easily link to search indexed headings.
- Heading embeds, complementing page embeds.
- Converting headings into pages.
- Be able to drag headings to rearrange them.
- Ability to zoom in on headings.
- page views/lens.
- And for further functionality down the line.

Would this be hard to implement given Tangent's existing page-first architecture?

**Comment by taylorhadden** [comment]
This generally aligns with where I've wanted the data structure of Tangent to go. I'll note that linking to headers is already right now with the `[[Note Name#Header Name]]` syntax (though there is no automatic handling of renames at the moment...). Right now that means you need to know the name of the note that the header is in, but that could be changed in the future.

The behavior you've sketched out here with tags, properties, folding, drag re-arranging, etc. are all things that have been on the to-do list for a long while. They've not been at the _top_ of the list, but they're there.

Converting headings to pages would fall under a bucket that I've been thinking about as "note refactoring" tools. I expect these to be an extension of (or very similar to) the Creation Rule system.

Can you speak more about what you're thinking about with regards to "zooming in" on headings and "page views/lens"?

**Reply by Thyverity** [reply]
Regarding zooming, it is a Logsec feature, basically it's focusing on block, and presenting it and its children as single page, you can zoom in recursively on children blocks. I suppose it helps to remain focused on a particular part of a large page.
As for page views, it is complementary and similar in function to folder/tags/queries views, it takes the headings and present them in alternate ways to a document layout, for example imagine a kanban board composed of headings for columns and their subheadings as entries in those columns, that's a kind of view you can have for pages.

**Reply by Thyverity** [reply]
With heading blocks I imagine the way you link to them can be improved by indexing the headings like you do with pages, they would show up on the auto-complete like pages do, streamlining the process of linking to headings and subheadings.

**Reply by taylorhadden** [reply]
All of those different views you describe sound like _excellent_ candidates for note lenses/views. None of those sound like things I wouldn't like to see.

With regards to headings being more obviously surfaced in autocomplete/search, I think that's a perfectly reasonable thing to turn into an issue!

**Comment by Thyverity** [comment]
Thinking about this, came to the conclusion that block structure is a unnecessary abstraction for the goals of Tangent, most if not all the features I described can be accomplished without the requirement of blocks being a thing.

Coming from Logsec where the block is crucial, I thought it was a necessary for the kind of function it provides, But it isn't the case for Tangent, where it isn't abstracted as Logsec.

**Comment by DutchPete** [comment]
> In both Obsidian and Tangent, the page is the smallest unit of information.

The way I understand Tangent as a newbie (Obsidian veteran) is that the zettelkasten concept plays an important role, and there atomicity is key, i.e. relatively short notes that fit on your screen in their entirety,

In such an environment header linking is of little or no value. Having said that, most of my notes are long to very long and headers are important for navigation, so I am happy that Tangent supports linking to headers.

Linking to blocks would be even nicer. Ex Roam Research users argue quite loudly to obtain block linking.

**Reply by taylorhadden** [reply]
Zettelkasten is not the driving concept, but it's certainly an inspiration. I want to keep the tool more flexible.

Linking to individual lines is an intriguing idea. I'm not _super_ happy with the "insert a random hash at the end of a line" implementation that Obsidian leans on. It feels like computer garbage in my notes.

**Comment by DutchPete** [comment]
> Zettelkasten is not the driving concept
Phew. 👍

> I want to keep the tool more flexible.
👍

> I'm not super happy with the "insert a random hash at the end of a line" implementation that Obsidian leans on.
I understand. But you'll need some sort of hook to link to, and what I like about the Obs system is that the hashes are easy to trace if you use a different app, whereas if you look at the Logseq hashes, which are something like 20 or 30 characters long and impossible to trace.
